File: Python/LedController.py
import RPi.GPIO as GPIO

class LedController:

  def __init__(self, pinNum, name):
    self.pin = pinNum
    self.name = name

    print("Initializing LED for {} on pin {}".format(self.name, self.pin))
    GPIO.setup(self.pin, GPIO.OUT)
    GPIO.output(self.pin, False)

  def turnOn(self):
    print( "    Turning {} LED on".format( self.name))
    GPIO.output(self.pin, True)

  def turnOff(self):
    print("    Turning {} LED off".format( self.name))
    GPIO.output(self.pin, False)#

  def teardown(self):
    print("Tearing down {} LED".format( self.name))
    self.turnOff()

File: Javascipt/ledController.js
var gpio = require('rpi-gpio');


//
// LED Controller
//
function ledController() {
	this.init = function (led, callback) {
		var self = this;
		console.log("Initializing LED for %s on pin %d", led.Name, led.Pin);
		gpio.setup(led.Pin, gpio.DIR_OUT, function() {
			self.turnOff(led, callback);
		});
	};

	this.turnOn = function (led, callback) {
		console.log( "    Turning %s LED on", led.Name);
		gpio.write(led.Pin, true, callback);
	};

	this.turnOff = function (led, callback) {
		console.log( "    Turning %s LED off", led.Name);
		gpio.write(led.Pin, false, callback);
	};

	this.teardown = function (led, callback){
		var self = this;
		console.log("Tearing down %s LED", led.Name);
		self.turnOff(led, callback);
	};
};

module.exports = new ledController;